src/type_checker/infer/call.rs

```diff
@@ -83,6 +83,13 @@ fn property_call(
             let (instance, property_mutable, raises) = match &inner_instance.node {
                 Node::Id { lit } => {
                     let field = instance.field(&lit, &property.pos)?;
+                    if field.private && field.in_class != env.state.in_class {
```

Comment:
Some code duplication due to sloppy implementation of recursion.
Should attempt to rewrite this file before merging if possible.